plugins/SlicerT/SlicerT.cpp

```diff
@@ -0,0 +1,666 @@
+
+		// exponential fade out, applyRelease kinda sucks
+		if (noteFramesLeft < m_fadeOutFrames.value())
+		{
+			for (int i = 0; i < frames; i++)
+			{
+				float fadeValue = (float)(noteFramesLeft - i) / m_fadeOutFrames.value();
+				// if the workingbuffer extends the sample
+				fadeValue = std::clamp(fadeValue, 0.0f, 1.0f);
+				fadeValue = pow(fadeValue, 2);
+
+				workingBuffer[i][0] *= fadeValue;
+				workingBuffer[i][1] *= fadeValue;
+			}
+		}
+
+		instrumentTrack()->processAudioBuffer(workingBuffer, frames + offset, handle);
+
+		// calculate absolute for the waveform
+		float absoluteCurrentNote = (float)currentNoteFrame / totalFrames;
+		float absoluteStartNote = (float)sliceStart / totalFrames;
+		float abslouteEndNote = (float)sliceEnd / totalFrames;
+		emit isPlaying(absoluteCurrentNote, absoluteStartNote, abslouteEndNote);
+	}
+	else { emit isPlaying(-1, 0, 0); }
+}
+
+// uses the spectral flux to determine the change in magnitude
+// resources:
+// http://www.iro.umontreal.ca/~pift6080/H09/documents/papers/bello_onset_tutorial.pdf
+void SlicerT::findSlices()
+{
+	if (m_originalSample.frames() < 2048) { return; }
+	m_slicePoints = {};
+
+	// computacion params
+	const int windowSize = 512;
+	const float minBeatLength = 0.05f; // in seconds, ~ 1/4 length at 220 bpm
+
+	int sampleRate = m_originalSample.sampleRate();
+	int minDist = sampleRate * minBeatLength;
+
+	// copy vector into one vector, averaging channels
+	float maxMag = -1;
+	std::vector<float> singleChannel(m_originalSample.frames(), 0);
+	for (int i = 0; i < m_originalSample.frames(); i++)
+	{
+		singleChannel[i] = (m_originalSample.data()[i][0] + m_originalSample.data()[i][1]) / 2;
+		maxMag = std::max(maxMag, singleChannel[i]);
+	}
+
+	// normalize
+	for (int i = 0; i < singleChannel.size(); i++)
+	{
+		singleChannel[i] /= maxMag;
+	}
+
+	// buffers
+	std::vector<float> prevMags(windowSize / 2, 0);
+	std::vector<float> fftIn(windowSize, 0);
+	fftwf_complex fftOut[windowSize];
+
+	fftwf_plan fftPlan = fftwf_plan_dft_r2c_1d(windowSize, fftIn.data(), fftOut, FFTW_MEASURE);
+
+	int lastPoint = -minDist - 1; // to always store 0 first
+	float spectralFlux = 0;
+	float prevFlux = 0;
+	float real, imag, magnitude, diff;
+
+	for (int i = 0; i < singleChannel.size() - windowSize; i += windowSize)
+	{
+		// fft
+		memcpy(fftIn.data(), singleChannel.data() + i, windowSize * sizeof(float));
+		fftwf_execute(fftPlan);
+
+		// calculate spectral flux in regard to last window
+		for (int j = 0; j < windowSize / 2; j++) // only use niquistic frequencies
+		{
+			real = fftOut[j][0];
+			imag = fftOut[j][1];
+			magnitude = sqrt(real * real + imag * imag);
+
+			// using L2-norm (euclidean distance)
+			diff = sqrt(pow(magnitude - prevMags[j], 2));
+			spectralFlux += diff;
+
+			prevMags[j] = magnitude;
+		}
+
+		// detect increases in flux
+		if (spectralFlux / prevFlux > 1.0f + m_noteThreshold.value() && i - lastPoint > minDist)
```

Comment:
prevFlux can be 0 here so you get division with 0. This is with fpe debug flags set.

```gdb
Thread 1 "lmms" received signal SIGFPE, Arithmetic exception.
0x00007fff7915072a in lmms::SlicerT::findSlices (this=0x7ffff28c0080) at /home/zonkmachine/builds/lmms/plugins/SlicerT/SlicerT.cpp:461
461			if (spectralFlux / prevFlux > 1.0f + m_noteThreshold.value() && i - lastPoint > minDist)
(gdb) bt full
#0  0x00007fff7915072a in lmms::SlicerT::findSlices() (this=0x7ffff28c0080) at /home/zonkmachine/builds/lmms/plugins/SlicerT/SlicerT.cpp:461
        i = 0
        windowSize = 512
        minBeatLength = 0.0500000007
        sampleRate = 44100
        minDist = 2205
        maxMag = 0.721549094
...
        spectralFlux = 426.011841
        prevFlux = 0
        real = 0.0628787279
        imag = -0.0043129921
        magnitude = 0.0630264729
        diff = 0.0630264729
        noteSnap = 32767
        timeSignature = 8847608
        samplesPerBar = 16711739
        sliceLock = 32767
```

Reply:
fixed